code/modules/antagonists/cult/cult.dm

```diff
@@ -215,7 +234,52 @@
 	var/cult_vote_called = FALSE
 	var/mob/living/cult_master
 	var/reckoning_complete = FALSE
-
+	var/cult_risen = FALSE
+	var/cult_ascendent = FALSE
+	
+/datum/team/cult/proc/check_size()
+	if(cult_ascendent)
+		return
+	var/alive = 1
+	var/cultplayers = 0
+	for(var/I in GLOB.player_list)
+		var/mob/M = I
+		if(M.stat != DEAD)
+			if(iscultist(M))
+				++cultplayers
+			else
+				++alive
+	var/ratio = cultplayers/alive
+	if(ratio > CULT_RISEN && !cult_risen)
+		for(var/datum/mind/B in members)
+			if(B.current)
+				SEND_SOUND(B.current, 'sound/hallucinations/i_see_you2.ogg')
+				to_chat(B.current, "<span class='cultlarge'>The veil weakens as your cult grows, your eyes begin to glow...")
+				addtimer(CALLBACK(src, .proc/rise, B.current), 200)
+		cult_risen = TRUE
+		
+	if(ratio > CULT_ASCENDENT && !cult_ascendent)
+		for(var/datum/mind/B in members)
+			if(B.current)
+				SEND_SOUND(B.current, 'sound/hallucinations/im_here1.ogg')
+				to_chat(B.current, "<span class='cultlarge'>Your cult is ascendent and the red harvest approaches - you cannot hide your true nature for much longer!!")
+				addtimer(CALLBACK(src, .proc/ascend, B.current), 200)
+		cult_ascendent = TRUE
+		
+		
+/datum/team/cult/proc/rise(cultist)
```

Comment:
suggestion: 1 proc for both rise/ascend with a second argument (there is not ever a time you will ascend without rising, but you will always rise before ascending)

Reply:
you could actually replace the proc with `try_to_rise(mob/living/cultist)` and put the checks in there, so the copypaste of `if(cult_team.cult_risen)` blah blah blah is not necessary.

Reply:
you're welcome to code this but if a maintainer doesn't have a problem with it i'd really rather leave it as is

Reply:
only problem with it currently is that there's the same copypasta thrown around so if someone wanted to add a tertiary item, or simply edit the current checks, they'd have to sift through multiple areas which gives avenues to bugs.

I don't mind free GBP tho :^)